src/core/main.qml

```diff
@@ -315,12 +333,18 @@ Window {
                     if(properties.enterItem.isDialog) {
                         return pushVTransition
                     } else {
+                        if(properties.enterItem.isMusicalActivity) {
+                            isBackgroundMusicEnabledInActivity = false
+                            backgroundMusic.pause()
+                        }
                         return pushHTransition
                     }
                 } else {
                     if(properties.exitItem.isDialog) {
                         return popVTransition
                     } else {
+                        main.isBackgroundMusicEnabledInActivity = true
+                        backgroundMusic.resume()
```

Comment:
can it be handled via the on...Changed signal instead to avoid ?

Reply:
onChanged signal of isBackgroundMusicEnabledInActivity?

Reply:
sure, done :)